### DISCUSSION TITLE
Faster Dev Server Boot

### DISCUSSION THREAD

**Discussion by mcrascal** [evidence-dev/evidence#2408] [General]
npm run dev takes _a lot_ longer to get to a live app than a bare sveltekit app does. In addition to #922, improving this would do a lot for the initial experience of trying out evidence, and for my own day to day experience of using it.

**Comment by csjh** [comment]
Some profiling suggests a major cause is large files in dependencies taking up a lot of time in a vite function called `ssrTransformScript` (~4.1s). Main perpetrators:

![image](https://github.com/evidence-dev/evidence/assets/61282104/7984b2c1-ce37-4bda-ab35-fc2814c1e773)
![image](https://github.com/evidence-dev/evidence/assets/61282104/567ddc08-2760-43eb-a42d-57573649e141)
![image](https://github.com/evidence-dev/evidence/assets/61282104/86c1c984-0a7e-4e32-ae13-b3adda46bbc3)


There's a few other fat time sinks like svelte file compilation (~1.7s) - 170 files are compiled when the dev server is started, so we should figure out if there's a way to avoid that.

**Comment by archiewood** [comment]
Can't actually see these images, `Non-Image content-type returned ` but I assume they are inadvertent additions?

**Comment by csjh** [comment]
Whoops, updated

**Comment by hughess** [comment]
I'm looking through vite config for something else and stumbled upon this warmup option - sharing in case it's useful here:
https://vitejs.dev/config/server-options.html#server-warmup

**Comment by csjh** [comment]
Yeah seems like that helps move the heavy bits to the start of the dev server instead of later in, but the end result is the same

**Comment by archiewood** [comment]
@csjh what was your rough profiling setup to get these measurements?

**Comment by csjh** [comment]
@archiewood I've been modifying `node_modules/vite/bin/vite.js` because their `--profile` flag wasn't working 100% correctly for me

Basically just go in that file and replace
```js
  session.post('Profiler.enable', () => {
    session.post('Profiler.start', start)
  })
```
(located near the bottom) with
```js
  session.post('Profiler.enable', () => {
    session.post('Profiler.start', start)
	setTimeout(() => {
		session.post('Profiler.stop', async (err, { profile }) => {
			if (err) {
				console.error(err)
				process.exit(1)
			}
			const fs = await import('node:fs')
			const path = await import('node:path')
			const filename = path.resolve(process.cwd(), 'v8-profile.cpuprofile')
			fs.writeFileSync(filename, JSON.stringify(profile))
			console.log(`Profile written to ${filename}`)
			process.exit(0)
		})
	}, 20000);
  })
```
and run vite with the `--profile` flag
after the `.cpuprofile` can be loaded into chrome
